File: service/src/main/java/top/emilejones/hhu/entity/FileNode.java
package top.emilejones.hhu.entity;

public class FileNode {
    private String elementId;
    private String fileName;

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getElementId() {
        return elementId;
    }

    public void setElementId(String elementId) {
        this.elementId = elementId;
    }

    @Override
    public String toString() {
        return "FileNode{" +
                "elementId='" + elementId + '\'' +
                ", fileName='" + fileName + '\'' +
                '}';
    }
}

File: app/src/main/kotlin/top/emilejones/hhu/domain/dto/FileNode.kt
package top.emilejones.hhu.domain.dto

data class FileNode (
    val fileName: String
) {
    private val childList: MutableList<TextNode> = ArrayList()

    fun addChild(childNode: TextNode) {
        childList.add(childNode)
    }

    fun getChild(index: Int): TextNode {
        return childList[index]
    }

    fun childNum(): Int {
        return childList.size
    }
}